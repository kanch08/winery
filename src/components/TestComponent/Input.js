import React from 'react';

const Input = props => {
    const {type, text, handleChange } = props;
        return (
            <div>
                <input
                    type={type}
                    placeholder={text}
                    onChange={handleChange}
                />
            </div>
        );
}
export default Input;

