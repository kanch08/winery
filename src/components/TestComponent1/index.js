import React from 'react';

const TestComponent = () => {
    const [count, setCount] = React.useState(0);

    return (
        <React.Fragment>
            <h3>{count}</h3>
            <span>
      <button id="count-up" type="button" onClick={() => setCount(count + 1)}>Count Up</button>
      <button id="count-down" type="button" onClick={() => setCount(count - 1)}>Count Down</button>
      <button id="zero-count" type="button" onClick={() => setCount(0)}>Zero</button>
    </span>
        </React.Fragment>

);
}

export default TestComponent;