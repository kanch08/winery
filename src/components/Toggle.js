import React, { useState } from 'react';

const Toggle = () => {
    const [toggled, setToggled] = useState(false);
    const toggle = () => setToggled(!toggled);
    return (
        <button onClick={toggle}>
            {toggled ? 'Toggled' : 'Toggle'}
        </button>
    )
}

export default Toggle;

// import React from 'react';
//
// class Toggle extends React.Component {
//     state = { toggled: false };
//
//     constructor(props) {
//         super(props)
//         this.toggle = this.toggle.bind(this);
//     }
//
//     toggle() {
//         this.setState({ toggled: !this.state.toggled });
//     }
//
//     render() {
//         const { toggled } = this.state;
//
//         return (
//             <button onClick={this.toggle}>
//                 {toggled ? 'Toggled' : 'Toggle'}
//             </button>
//         )
//     }
// }
//
// export default Toggle;
