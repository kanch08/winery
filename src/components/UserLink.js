import React from 'react';
import { withRouter } from 'react-router';

const UserLink = ({ history }) => (
    <button onClick={() => history.push('/customer')}>Show all users</button>
);

export default withRouter(UserLink);
