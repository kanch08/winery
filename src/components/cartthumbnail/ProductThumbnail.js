import React, {Component} from 'react';
import CartMenuOption from './CartMenuOption';
import {connect} from 'react-redux';
import Pulse from 'react-reveal/Pulse';
import {Link} from "react-router-dom";

class ProductThumbnail extends Component {
    constructor(props) {
        super(props);
        this.state = {
            cartMenuRender: false,
            selectedImage: this.props.defaultImage,
        }
    }

    cartMenuOnOver = (event) => {
        event.stopPropagation();
        this.setState({
            cartMenuRender: !this.state.cartMenuRender,
        })
    };

    handleChange = (e) => {
        console.log("radio button change func called with event", e);
        this.setState({
            selectedImage: e.target.value
        })
    }

    render() {
        const {productname, price, alt, id} = this.props;
        let imageRender = this.props.images;
        let imageKeys = Object.keys(this.props.images[0]);

        return (
            <Pulse>
                <div className="productThumbnail"
                     onMouseEnter={this.cartMenuOnOver}
                     onMouseLeave={this.cartMenuOnOver}
                >

                    {
                        imageKeys.map(item => {
                                return (
                                    this.state.selectedImage === item ?
                                        <Link className="link" to={{pathname:`productdetails/${id}`}} key={id}  ><div className="image">
                                            <img src={imageRender[0][item]} alt={alt}/>
                                        </div></Link> : null

                                )
                            }
                        )
                    }

                    <div className="detail">
                        <p>{productname}</p>
                        <p>{'\u20B9'}{price}</p>
                    </div>
                    <div className={`${productname} buttonNav`}>
                        {
                            this.state.cartMenuRender ?
                                <CartMenuOption
                                    id={id}
                                    onRadioChange={this.handleChange}
                                    imageKeys={imageKeys}
                                    name={productname}
                                    price={price}
                                    currentColor={this.state.selectedImage}
                                /> : null
                        }
                    </div>

                </div>
            </Pulse>
        )
    }

}

const mapStateToProps = (state) => {
    return {
        images: state.reducer.images,
    }
}
export default connect(mapStateToProps, '')(ProductThumbnail);