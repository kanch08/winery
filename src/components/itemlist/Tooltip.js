import React, {Component} from 'react';

class Tooltip extends Component {
    render() {
        return (
            <div className="tooltip">
                <p>Quantity must not be greater than 10</p>
            </div>
        );
    }
}

export default Tooltip;