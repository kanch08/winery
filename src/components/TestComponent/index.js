import React, { Component } from 'react';
import Input from './Input'

class TestComponent extends Component {
    constructor(props) {
        super(props);
        this.state = {count: 0, name: ''};
    }

    handleChange = (e) => {
        console.log("e", e.target);
        this.setState({name: e.target.value});
    }

    render() {
        const {name} = this.state;
        console.log("name", name)
        return (
            <React.Fragment>
                <h3>{this.state.count}</h3>
                <span>
        <button id="count-up" type="button" onClick={() => this.setState({count: this.state.count + 1})}>
          Count Up!
        </button>
        <button id="count-down" type="button" onClick={() => this.setState({count: this.state.count - 1})}>
          Count Down!
        </button>
        <button id="zero-count" type="button" onClick={() => this.setState({count: 0})}>Zero</button>
      </span>
                <Input
                    type="text"
                    placeholder="enter your name"
                    value={name}
                    onChange={(e) => this.handleChange(e)}
                    />
            </React.Fragment>
        )
    }
}

export default TestComponent;