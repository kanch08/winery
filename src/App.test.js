import React from 'react'
import { shallow } from 'enzyme'
import List from "./components/List";
import Paragraph from "./components/Paragraph";
import Toggle from "./components/Toggle";
import Tooltip from "./components/itemlist/Tooltip";
import UserLink from "./components/UserLink";
import ProductThumbnail from "./components/cartthumbnail/ProductThumbnail";
import toJson from "enzyme-to-json";
import TestComponent from "./components/TestComponent/index";
import TestComponent1 from "./components/TestComponent1/index";

describe('Paragraph' , () => {
  it('should render children inside a p tag',  () => {
    const wrapper = shallow(<Paragraph>This is my first test</Paragraph>);
    const paragraph = wrapper.find('p');
    expect(paragraph).toHaveLength(1);
    expect(paragraph.text()).toEqual('This is my first test')
  })
})


describe('List'  , () => {
  it('should render This list is empty inside a paragraph component if items list is an empty array' , () => {
    const wrapper = shallow(<List items={[]} />);
    const paragraph = wrapper.find(Paragraph);
    expect (paragraph.props().children).toEqual('This list is empty');
  })

  it('should render a list with the body of each item inside a div', () => {
    const itemsArray = [{ id: 1, body: 'Shopping' }, { id: 2, body: 'Exercise' }, { id: 3, body: 'Cook' }]
    const wrapper = shallow(<List items={itemsArray} />)
    const items = wrapper.find('div[className="item"]')
    expect(items).toHaveLength(itemsArray.length)
    expect(items.first().text()).toEqual('Shopping')
    console.log(wrapper.debug());
  })
})

describe('Toggle' , () => {
  describe('Behavioral(Integration', () => {
    const wrapper = shallow(<Toggle />);
    it('renders a button with "Toggle" as children', () => {
     expect(wrapper.find('button')).toHaveLength(1)
     expect(wrapper.find('button').text()).toEqual('Toggle')
    })

    it('renders "Toggled" as button children if button is clicked' , () => {
      wrapper.find('button').simulate('click');
      expect(wrapper.find('button').text()).toEqual('Toggled')
    })

    it('renders "Toggle" as button children if button is clicked' , () => {
      wrapper.find('button').simulate('click');
      expect(wrapper.find('button').text()).toEqual('Toggle')
    })
  })

  describe('Component (Unit)', () => {
    const wrapper = shallow(<Toggle />)
   describe('Toggle function' , () => {
     it('toggles "toggled" variable in state' , () => {
       expect(wrapper.state('toggled')).toBe(false)
       wrapper.instance().toggle();
       expect(wrapper.state('toggled')).toBe(true)
     })
   })
  })
})

const history = {
  push: jest.fn(),
};

describe('it renders enzyme' , () => {
  it('testing Paragraph', () => {
    const wrapper = shallow(<Paragraph/>)
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})

describe('testing the Test Component', () => {
  const wrapper = shallow(<TestComponent/>)
  it('has the initial state count of zero', () => {
    expect(wrapper.state()).toEqual({ count: 0, name: ''});
  })

  it('increments by 1 on one click', () => {
    wrapper.find('#count-up').props().onClick();
    expect(wrapper.state()).toEqual({ count: 1, name: ''});
  })

  it('decrements by 1 on one click', () => {
    wrapper.find('#count-down').props().onClick();
    expect(wrapper.state()).toEqual({ count: -1, name: ''});
  })

  it('sets state count to 0 on click', () => {
    wrapper.setState({count: 10})
    wrapper.find('#zero-count').props().onClick();
    expect(wrapper.state()).toEqual({count: 0, name: ''});
  })
})

describe('hook component useState', () => {
  let wrapper;
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState')
  useStateSpy.mockImplementation((init) => [init, setState]);

  beforeEach(() => {
    wrapper = shallow(<TestComponent1 />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('Count Up', () => {
    it('calls setCount with count + 1', () => {
      wrapper.find('#count-up').props().onClick();
      expect(setState).toHaveBeenCalledWith(1);
    });
  });

  describe('Count Down', () => {
    it('calls setCount with count - 1', () => {
      wrapper.find('#count-down').props().onClick();
      expect(setState).toHaveBeenCalledWith(-1);
    });
  });

  describe('Zero', () => {
    it('calls setCount with 0', () => {
      wrapper.find('#zero-count').props().onClick();
      expect(setState).toHaveBeenCalledWith(0);
    });
  });
});