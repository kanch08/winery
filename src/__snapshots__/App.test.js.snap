// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders enzyme testing Paragraph 1`] = `<p />`;
